src/core/geometry/qgswkbptr.h

```diff
@@ -26,33 +37,50 @@
 class CORE_EXPORT QgsWkbPtr
 {
     mutable unsigned char *mP;
+    unsigned char *mStart;
+    unsigned char *mEnd;
+
+    void verify( int size ) const { Q_ASSERT( mP && mP + size <= mEnd ); if ( !mP || mP + size > mEnd ) throw QgsWkbException( "wkb access out of bounds" ); }
```

Comment:
Should Q_ASSERT be removed here ?
You can "catch throw" under gdb, if that's the aim of the assertion...